### ISSUE TITLE
all links opened in the same container?

### ISSUE THREAD

**Issue by gliptak** [mozilla/multi-account-containers#914]
Consider the following usecase:

+ open https://www.bankofamerica.com/ in a "banking container" tab
+ scroll to the bottom of the page and click on the Youtube icon and https://www.youtube.com/user/bankofamerica opens in a new tab
+ this new tab is also opened in a "banking container"

For isolation, I would expect Youtube to be opened in "default" (or its assigned) container to be unaware of my banking context details.

Is there a (per container?) setting I'm not aware of to make links open in "default container"?

Thanks

**Comment by MikkCZ** [comment]
I wonder if this should be part of the mozilla add-on, or a space for add-ons like Conex and others.

**Comment by gliptak** [comment]
For reference

https://hacks.mozilla.org/2017/10/containers-for-add-on-developers/

https://addons.mozilla.org/en-US/firefox/collections/jkt/container-tab-addons/

And in particular

https://addons.mozilla.org/en-US/firefox/addon/containers-on-the-go/

https://addons.mozilla.org/en-US/firefox/addon/conex/?

**Comment by MikkCZ** [comment]
Btw. the containers would probably need to go beyond a tab, e.g. embedded YouTube videos or any remote resources.

**Comment by gliptak** [comment]
@MikkCZ yes, it would be great if we would hear from Mozilla devs on this. From an end-user perspective, the current default behavior doesn't seem to isolate right ...

**Comment by MikkCZ** [comment]
In certain cases you want the current behavior (not to break websites etc.). Also keeping a domain strictly in a single container won't allow you to login with multiple accounts. In terms of the YouTube example, you may (or may not) to save videos seen on all websites from all containers into your YouTube account history. This is extremely hard to set up, it's a question on user what he/she prefers and will also differ per domain or website, maybe per particular case.

Also what about cdn and similar embedded sources, that may track you across websites thanks to your unique settings and the way you serve or not serve cookies to various domains during a single page load.

**Comment by biva** [comment]
The default behavior could be discussed, but I'd like to be able to chose if opening a link in a new tab should be opened in the same container, or in the default one.
Or a keyboard option should be available (for example Ctrl + middle click = open in a new default tab)

**Comment by gliptak** [comment]
@MikkCZ So maybe same (sub)domain same container, and domains with their assigned container in those containers (I cannot speak to how/why this would brake websites).
@biva And a setting to default open link to "default" container or to same container.

**Comment by cole-h** [comment]
@gliptak I haven't tested it myself, but supposedly [Containerise](https://github.com/kintesh/containerise) can do this through its wildcarding feature. Might be worth checking out.

**Comment by gliptak** [comment]
@cole-h Thank you for the pointer

**Comment by jonathanKingston** [comment]
I agree, this is kinda the inverse of assignment. This is almost a dupe of #847 too.

**Comment by gliptak** [comment]
I'm not sure how wildcards would work with [Containerize](https://github.com/kintesh/containerise/) but this [issue](https://github.com/kintesh/containerise/issues/10) details functionality I would like to have available
